### PR TITLE
Fallback to oc command if kubectl is not found

### PR DESCRIPTION
**Pull request by cdrage**
This checks to see if the kubectl and oc commands are available and
fallback to the oc command appropriatley if deploying a container
without kubectl installed on the host system.

Closes https://github.com/kedgeproject/kedge/issues/458

### CODE REVIEW

**Comment by cdrage** [comment]
Not too sure how to make integrations tests for this, but :man_shrugging:

**Comment by cdrage** [comment]
@kadel @surajnarwade @containscafeine review?

File: pkg/cmd/commands.go [hunk 1]
```diff
@@ -131,17 +132,38 @@ func CreateArtifacts(paths []string, generate bool, args ...string) error {
...
+	if executable == "kubectl" {
+		if _, err := exec.LookPath("kubectl"); err != nil {
+			log.Debug("kubectl is unavailable, using oc")
+			executable = "oc"
+		}
+	}
+
+	// If oc is used, error out if it's not available
+	if executable == "oc" {
+		if _, err := exec.LookPath("oc"); err != nil {
```
[the remark this answers is not in the document]
Comment (cdrage) [reply pkg/cmd/commands.go]: @kadel tried, but combining both doesn't work especially when assigning new variables :unamused:

File: pkg/cmd/commands.go [hunk 2]
```diff
@@ -131,17 +132,38 @@ func CreateArtifacts(paths []string, generate bool, args ...string) error {
...
 func RunClusterCommand(args []string, data []byte, useOC bool) error {
+
+	// Use kubectl by default, oc if useOC bool is true (in cases such as DeploymentConfig, ImageStream, etc.)
 	executable := "kubectl"
 	if useOC {
 		executable = "oc"
 	}
+
+	// Use oc if kubectl is unavailable
+	if executable == "kubectl" {
```
[the remark this answers is not in the document]
Comment (cdrage) [reply pkg/cmd/commands.go]: Yup. I'll change this.

**Comment by cdrage** [comment]
Thanks @kadel I've made the changes and pushed it again :100:

**Comment by pradeepto** [comment]
@containscafeine Can you have a look at this? @kadel is on PTO from tomorrow until Jan 2. I would like to see this in master and in the next release because other teams have asked for this.  Thanks.

**Comment by surajnarwade** [comment]
Worked for me, LGTM

**Pull request merged**